Passwordless Auth - Enable passwordless sign up for the core profiler flow

moon0326 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #92104 

## Proposed Changes

* Enable passwordless sign up
* Applies new design changes
* Email template change is not implemented yet (Image 3)

## Why are these changes being made?

* To enable new design changes and enable passwordless signup

## Testing Instructions

This PR requires https://github.com/woocommerce/woocommerce-start-dev-env environment as it requires sign up process.

1. Setup https://github.com/woocommerce/woocommerce-start-dev-env 
2. Once https://github.com/woocommerce/woocommerce-start-dev-env is ready, run `yarn start` in `wp-calypso`
3. Open a new incognito window
4. Create a new JN site with the latest WooCommerce
5. Go through the core profiler and make sure to select Jetpack on the plugins page.
6. Once you get to the jetpack connect page, confirm the new design (Image 1)
7. Enter an email address and click on the `Continue` button.
8. You should logged-in and see `continue as` design (Image 2)

Image 1:
![Screenshot 2024-08-02 at 5 32 29 PM](https://github.com/user-attachments/assets/a0391245-7c33-4a88-987d-67228e92cff6)

Image 2:
![Screenshot 2024-08-02 at 5 30 55 PM](https://github.com/user-attachments/assets/bb4cf562-e778-4ef2-81f9-4f678c99a210)

Image 3:
![image](https://github.com/user-attachments/assets/0edb55e1-17f0-4d71-822f-d687e5539013)




## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Enable passwordless sign up form for the core profiler

Commit:
Change prop name: hideSiteCard to disableSiteCard

Commit:
Add disableTosText prop to control the visibility of the tos text

Commit:
Core profiler passwordless css

Commit:
Use username from the API response when it is not provided in the payload

Commit:
Treat woo core profiler as woo passwordless

chihsuan commented on client/state/selectors/get-is-woo-passwordless.ts:
```diff
@@ -14,6 +14,11 @@ export default function getIsWooPasswordless( state: AppState ): boolean {
 	}
 
+	// Enable Woo Passwordless if user is from WooCommerce Core Profiler.
+	if ( state?.route?.query?.initial?.from === 'woocommerce-core-profiler' ) {
```
We can reuse `isWooCommerceCoreProfilerFlow` selector here.

https://github.com/Automattic/wp-calypso/blob/eef6730df07f7816be791932cfbab18184731021/client/state/selectors/is-woocommerce-core-profiler-flow.ts#L12-L17

moon0326 replied:
Thank you! Updated in [use isWooCommerceCoreProfilerFlow](https://github.com/Automattic/wp-calypso/pull/93230/commits/3a09af66d597209a263a704430f80fb325522c4e)

chihsuan reviewed (commented):
Thanks for working on this! 

I tested the flow and found some UI issues:

### Sign up screen

![Screenshot 2024-08-06 at 13 54 30](https://github.com/user-attachments/assets/bb95d214-2185-477a-99bd-aee53447d468)


1. "Log in" link color is purple, but it should be blue 
2. The padding between heading and form is too much
3. The email label should be `your email` instead of `enter you email address`
4. The font family of the social button (Inter) is different from the design (SF pro) but I think we should confirm with Veronica on this. It would be great if we can use the same font family as Woo.com signup flow. Similarly, the font size of the heading is also different from the design.

It seems that some issues are because the CSS is not applied properly. I think maybe we should move styles in the .woocommerce-core-profiler to is-woocommerce-core-profiler-flow.is-woo-passwordles to avoid conflicts with other styles.


### Login screen

![Screenshot 2024-08-06 at 14 09 36](https://github.com/user-attachments/assets/d2559223-f1ce-4ea6-840f-dceca1eda804)


1. The TOS text has underline in the design but it doesn't have it in the implementation. I think we also confirmed with Veronica. It looks weird with underline.
2. When I use a account that with a password, it shows the password field but also shows `If you don’t have an account, we’ll use this email to create it.` message. I think we should hide message or confirm if it's expected behavior.
3. Click on `Signup up" link, it shows flash of the old design before redirecting to signup page. I think we should fix this.
4. The Lost password link redirects to WP lost password page. I think we should redirect to Woo-branded lost password page.

https://github.com/user-attachments/assets/ea91a3cc-d291-482d-8a23-570be9f211f7

https://github.com/user-attachments/assets/74e404a7-0b81-481f-b81d-f60d78b17a19

chihsuan commented on client/state/jetpack-connect/actions/create-account.js:
```diff
@@ -27,6 +27,8 @@ export function createAccount( userData ) {
 				client_id: config( 'wpcom_signup_id' ),
 				client_secret: config( 'wpcom_signup_key' ),
 			} );
```
It might be worth checking user data to see if there are any fields that can be used to customize the email template.

moon0326 commented:
Thank you for the detailed review @chihsuan 🙇 

I'm going to address any signup-related issues in this PR, but I'll create a new PR for the login-related issues. This PR is only for signup.

Commit:
use isWooCommerceCoreProfilerFlow

Commit:
Use theme color for the login link

Commit:
Use correct input label

Commit:
Update subtitle text copy

Commit:
Adjust styles

* Removed extra margin between the subtitle and the signup form
* Updated font-size for the input and its label

moon0326 commented:
@chihsuan 

> "Log in" link color is purple, but it should be blue

Updated in [Use theme color for the login link](https://github.com/Automattic/wp-calypso/pull/93230/commits/0e66660c779073aaa574b635561f59ed0b290205)

> The padding between heading and form is too much

Updated in [Adjust styles](https://github.com/Automattic/wp-calypso/pull/93230/commits/e07ed0532189b75844a1dca2c8ac5359d50c7be7)

> The email label should be your email instead of enter you email address

Updated in [Use correct input label](https://github.com/Automattic/wp-calypso/pull/93230/commits/ca47549cb59736fc8183f6f2417a915ac286b493)


> The font family of the social button (Inter) is different from the design (SF pro) but I think we should confirm with Veronica on this. It would be great if we can use the same font family as Woo.com signup flow. Similarly, the font size of the heading is also different from the design.

I'll handle this in a separate PR as it applies to all. I'll need to see how Woo is implemented.  I'll also address login page related feedback in a separate PR.

chihsuan reviewed (approved):
Thanks for addressing the feedback! @moon0326  Code looks good to me and confirmed that I can create a new account.

Just one thing, after I clicked on "Continue with email" button" and entered the loading state, the screen showed wordpress loading screen in the middle. I think that is a bit weird. We probably need to look into this but I think it's out of scope for this PR.


https://github.com/user-attachments/assets/40e691a8-d7d8-4c88-87a3-0a62c0dde968

I'll approve this PR now. Thanks for your work on this! 🎉

moon0326 commented:
> Just one thing, after I clicked on "Continue with email" button" and entered the loading state, the screen showed wordpress loading screen in the middle. I think that is a bit weird. We probably need to look into this but I think it's out of scope for this PR.

Thank you! So the first loading is from `signup.js` and the other one is coming from `authorize.js`. They are from different pages. I think we need to find a way to hide the WP loading screen. I'll look into that in a separate PR.

Thank you for the review 🙇
